plasmapy/dispersion/numerical/hirose_.py

```diff
@@ -0,0 +1,315 @@
+"""
+This module contains functionality for calculating various numerical
+solutions to Hirose's two fluid dispersion relation, see
+:cite:t:`hirose:2004` and :cite:t:`bellan:2012`.
+"""
+
+import astropy.units as u
+import numpy as np
+import warnings
+
+from astropy.constants.si import c
+from typing import Union
+
+from plasmapy.formulary.frequencies import gyrofrequency, plasma_frequency
+from plasmapy.formulary.speeds import Alfven_speed, ion_sound_speed
+from plasmapy.particles import Particle
+from plasmapy.particles.exceptions import ChargeError
+from plasmapy.utils.decorators import validate_quantities
+from plasmapy.utils.exceptions import PhysicsWarning
+
+c_si_unitless = c.value
+
+
+@validate_quantities(
+    B={"can_be_negative": False},
+    n_i={"can_be_negative": False},
+    T_e={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+)
+def hirose(
+    *,
+    B: u.T,
+    ion: Union[str, Particle],
+    k: u.rad / u.m,
+    n_i: u.m**-3,
+    T_e: u.K,
+    theta: u.rad,
+    gamma_e: Union[float, int] = 1,
+    gamma_i: Union[float, int] = 3,
+    z_mean: Union[float, int] = None,
+    **kwargs,
+):
+
+    r"""
+    Calculate the two fluid dispersion relation presented by
+    :cite:t:`hirose:2004`, and discussed by :cite:t:`bellan:2012`.
+    This is a numerical solver of equation 7 in :cite:t:`bellan:2012`.
+    See the **Notes** section below for additional details.
+
+    Parameters
+    ----------
+    B : `~astropy.units.Quantity`
+        The magnetic field magnitude in units convertible to T.
+    ion : `str` or `~plasmapy.particles.particle_class.Particle`
+        Representation of the ion species (e.g., ``'p'`` for protons, ``'D+'``
+        for deuterium, ``'He-4 +1'`` for singly ionized helium-4, etc.). If no
+        charge state information is provided, then the ions are assumed to be
+        singly ionized.
```

Comment:
```suggestion
        Representation of the ion species (e.g., ``"p"`` for protons,
        ``"D+"`` for deuterium, ``"He-4 +1"`` for singly ionized
        helium-4, etc.). If no charge state information is provided,
        then the ions are assumed to be singly ionized.
```

Just updating to match how we've reworded this description in other doctrings, and pep8 line length.